base/compiler/ssair/passes.jl

```diff
@@ -446,7 +446,9 @@ function lift_arg!(
     end
     if isa(lifted, GlobalRef) || isa(lifted, Expr)
         lifted = insert_node!(compact, leaf, effect_free(NewInstruction(lifted, argextype(lifted, compact))))
+        compact[leaf] = nothing
         stmt.args[argidx] = lifted
+        compact[leaf] = stmt
```

Comment:
Here we forgot to insert, but we also must use the `compact[i] = nothing` pattern to avoid miscouting